From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Tue, 3 Oct 2023 15:00:45 +0800
Subject: layout: Destroy panel barrier on shutdown

It was being leaked, which is detected on mutter shutdown.

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/3011
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2975>
(cherry picked from commit 4e2dddd18ef27da768f8b0fc2c8e243a3b207f21)

Origin: upstream, after 45.0
---
 js/ui/layout.js | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/js/ui/layout.js b/js/ui/layout.js
index 06ed440..95e34f2 100644
--- a/js/ui/layout.js
+++ b/js/ui/layout.js
@@ -243,6 +243,7 @@ export const LayoutManager = GObject.registerClass({
             }
 
             this._destroyHotCorners();
+            this._destroyPanelBarrier();
             this.uiGroup.destroy();
         });
 
@@ -575,11 +576,15 @@ export const LayoutManager = GObject.registerClass({
         });
     }
 
-    _updatePanelBarrier() {
+    _destroyPanelBarrier() {
         if (this._rightPanelBarrier) {
             this._rightPanelBarrier.destroy();
             this._rightPanelBarrier = null;
         }
+    }
+
+    _updatePanelBarrier() {
+        this._destroyPanelBarrier();
 
         if (!this.primaryMonitor)
             return;
